### PR TITLE
[AzureSearch] Add POST method for ListQueryKeys operation

### PR DESCRIPTION
**Pull request by tjacobhi**

### CODE REVIEW

File: specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json [hunk 1]
```diff
@@ -274,6 +275,55 @@
...
         }
+      },
+      "post": {
+        "tags": [
+          "QueryKeys"
+        ],
+        "x-mspageable": {
+          "nextLinkName": null
+        },
+        "operationId": "QueryKeys_ListBySearchServicePost",
```
Comment (brjohnstmsft) [inline specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json]: That's not a great name for a public SDK method. Can we instead just remove the GET version of this from the Swagger and re-use the same name? Ideally the parameters wouldn't change on the client side so it wouldn't be a breaking change.

**Review by bitsforthought — CHANGES_REQUESTED** [review]

**Review by KrisBash — APPROVED** [review]
Discussed offline. Signing off for ARM

File: specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json (hunk 1 above)
Comment (tjacobhi) [reply specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json]: I thought we get S360 issue if we still have something reachable without it being in the swagger?

Comment (brjohnstmsft) [reply specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json]: Define "reachable". If we remove it from our RP manifest and ensure it doesn't show up in the Operations API, is it no longer reachable? I realize that is not an option for this API version, but we should definitely do that in future API versions.

You can leave it in the Swagger, but I'd recommend finding a way to exclude that operation from codegen, and using the old name (which is a good name) for the new POST-based operation.

Comment (tjacobhi) [reply specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json]: Any ideas of how I can prevent it from codegen? I marked it with a deprecated tag, but not sure how that impacts codegen.

Comment (brjohnstmsft) [reply specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json]: AutoRest options are documented [here](https://github.com/Azure/autorest/tree/master/docs/extensions). If you can't find one that does what you want, you can always use the big hammer of adding regex rules to the [AutoRest config file](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/search/resource-manager/readme.md). If it comes to that, you can look in the data plane configs for examples.

File: specification/search/resource-manager/Microsoft.Search/stable/2015-02-28/search.json [hunk 2]
```diff
@@ -118,6 +118,55 @@
             }
           }
         }
+      },
+      "post": {
```
Comment (brjohnstmsft) [inline specification/search/resource-manager/Microsoft.Search/stable/2015-02-28/search.json]: We're adding this to 2015-02-28 too?

File: specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json [hunk 3]
```diff
@@ -233,7 +233,7 @@
         "x-ms-pageable": {
           "nextLinkName": null
         },
-        "operationId": "QueryKeys_ListBySearchService",
+        "operationId": "QueryKeys_ListBySearchService_Get",
```
Comment (brjohnstmsft) [inline specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json]: I'm not sure how AutoRest will handle the underscore before `_Get`. If it doesn't do something sensible, just remove it.

**Review by bitsforthought — APPROVED** [review]

File: specification/search/resource-manager/Microsoft.Search/stable/2015-02-28/search.json (hunk 2 above)
Comment (tjacobhi) [reply specification/search/resource-manager/Microsoft.Search/stable/2015-02-28/search.json]: I noticed that it was implemented for all api versions, so I thought the documentation should match.

**Review by ChenTanyi — APPROVED** [review]

File: specification/search/resource-manager/Microsoft.Search/stable/2015-02-28/search.json [hunk 4]
```diff
@@ -127,11 +127,6 @@
           "nextLinkName": null
         },
         "operationId": "QueryKeys_List",
-        "x-ms-examples": {
```
Comment (brjohnstmsft) [inline specification/search/resource-manager/Microsoft.Search/stable/2015-02-28/search.json]: Every operation needs an example, or else validation fails, no?

**Review by bitsforthought — CHANGES_REQUESTED** [review]

File: specification/search/resource-manager/Microsoft.Search/stable/2015-02-28/search.json (hunk 2 above)
Comment (brjohnstmsft) [reply specification/search/resource-manager/Microsoft.Search/stable/2015-02-28/search.json]: It probably shouldn't be implemented for all API versions.

In general, we don't retroactively update Swagger for old API versions. No harm in doing it, it's just extra maintenance that doesn't usually need to be done.

**Comment by ChenTanyi** [comment]
@tjacobhi will you fix the CI error and the changes requested? Otherwise, I'll close the PR.

**Comment by brjohnstmsft** [comment]
@ChenTanyi Jacob is on a leave of absence. @miwelsh -- This PR needs a new owner.

**Review by jqhuangonearth — APPROVED** [review]

**Review by jqhuangonearth — APPROVED** [review]

**Review by jqhuangonearth — APPROVED** [review]

**Review by bitsforthought — APPROVED** [review]

**Comment by huangbolun** [comment]
@KrisBash , @ChenTanyi: could you merge the PR? I still have some unsuccessful checks which I don't know how to resolve. Could you suggest?

**Review by abhi1509 — APPROVED** [review]

**Comment by ChenTanyi** [comment]
@NullMDR Can you take a look at the sdk automation failure? Does it matter?

**Comment by NullMDR** [comment]
@ChenTanyi That's some race condition caused bug. You can ignore it.

**Pull request merged**